### ISSUE TITLE
Measuring performance with and without Service Worker

### ISSUE THREAD

**Issue by aashishbarnwal** [w3c/ServiceWorker#1462]
We are conducting an experiment to gauge the benefits of service worker. As part of that we are recording pageLoadTime and service worker state. The idea is that the data which has controller state as activated will be treated as pages served via service worker and where service worker is not installed will have controller as null and so no service worker state. This way we can compare pageLoadTime with & without service worker.

Is this the right approach to know that the page is being served from SW? One concern that comes to mind is, if SW gets killed in between and gives us wrong info about the state. However we are recording service worker state way early in page load lifecycle as soon as our js code boots up. So at this point SW should be up & running intercepting traffic from the page. This looks to be a safe technique. Any thoughts are this ?

Another approach that comes to mind after reading PerformanceResourceTiming API on MDN is to use workerTime.

> The workerStart read-only property of the PerformanceResourceTiming interface returns a DOMHighResTimeStamp immediately before dispatching the FetchEvent if a Service Worker thread is already running, or immediately before starting the Service Worker thread if it is not already running. If the resource is not intercepted by a Service Worker the property will always return 0.

As per the doc, it mentions workerTime will always be zero if SW is not there. Can there be a scenario when SW is installed and yet workerTime is zero?
As an trial, I crawled to Pinterest and queried following in Chrome dev console, workerTime is always zero though SW is there.

> performance.getEntriesByType('navigation')[0]

[Link to StackOverflow](https://stackoverflow.com/questions/57578604/measuring-performance-with-and-without-service-worker)

**Comment by wanderview** [comment]
Does pinterest have a fetch handler?  It will only be non-zero if the service worker is handling the FetchEvent.

Still I don't think workerStart is reliably non-zero across browsers in current implementations.

I have a proposal for the service worker to append additional data to the PerformanceResourceTiming entry that could maybe be used to address your use case here:

https://github.com/wanderview/fetchevent-worker-timing/blob/master/explainer.md

That proposal has not been implemented yet, though.

**Comment by aashishbarnwal** [comment]
@wanderview Thanks Ben for the response!
I do see fetch handler in Pinterest's SW impl. I came across your proposal to enrich PerformanceResourceTiming sometime back, but don't see it moving.

`self.addEventListener('fetch', handleFetch);`

**Comment by jakearchibald** [comment]
Closing this, as the question seems answered. Happy to keep following up, and create a new issue if there's work as a result of this.
